lib/azure/armrest/model/base_model.rb

```diff
@@ -22,6 +22,8 @@ def self.attr_hash(*attrs)
 
       attr_hash :tags
 
+      attr_accessor :response_headers
```

Comment:
The response_headers need to be public IMO so that people can inspect rate limit information.